ui-manchette/src/components/helpers.ts

```diff
@@ -93,3 +125,18 @@ export const getScales = (ops: OperationalPoint[], isProportional: boolean, zoom
   }));
   return scales;
 };
+
+/** Zoom on X axis and center on the mouse position */
+export const zoomX = (
+  currentXZoom: number,
+  currentXOffset: number,
+  { delta, position: { x } }: Parameters<NonNullable<SpaceTimeChartProps['onZoom']>>[0]
+) => {
+  const xZoom = Math.min(Math.max(currentXZoom * (1 + delta / 10), MIN_ZOOM_X), MAX_ZOOM_X);
+  return {
+    xZoom,
+    // Adjust zoom level relatively to the input delta value:
+    // These lines are here to center the zoom on the mouse position:
+    xOffset: x - ((x - currentXOffset) / currentXZoom) * xZoom,
+  };
+};
```

Comment:
Wasn't it possible to use the `zoom` function from spaceTimeChart ? (maybe with a little refacto to use only one of the two axis) ?

Reply:
No unfortunately, we cannot since we have separated modules and this function zoomX is not exported from ui-space-time-chart (on purpose)